pontus/week4/app/named-list.jsx

```diff
@@ -0,0 +1,88 @@
+let React = require('react');
+let Message = require('./message');
+let MessageActions = require('./message-actions');
+let MessageStore = require('./message-store');
+
+let NamedList = React.createClass({ 
+	propTypes: { 
+		listID: React.PropTypes.number,
+		listName: React.PropTypes.string,
+		messages: React.PropTypes.arrayOf(React.PropTypes.object),
+		listProperties: React.PropTypes.arrayOf(React.PropTypes.object)
+	},
+
+	getInitialState() {
+		return { 
+			charCount: 200,
+			isError: false,
+			errorMsg: "",
+			displayChars: false
+		};
+	},
+
+	createMsg(e) { 
+		if (e.keyCode == 13 || e.which == 13 || e.type == 'click') { 
+			let text = this.refs.inputField.value;
+
+			MessageActions.createMessage(this.props.listID, text);
+
+			this.refs.inputField.value = "";
+			this.refs.inputField.focus();
+		}
+	},
+
+	displayCharCount() { 
+		this.setState({ displayChars: !this.state.displayChars }); 
+	},
+
+	charValidation() { 
+		let charCount = 200 - this.refs.inputField.value.length;
+
+		if (charCount < 0) this.setState({ isError: true, errorMsg: "out of characters." });
+		else this.setState({ isError: false, errorMsg: "" }) 
+
+		this.setState({ charCount });
+	},
+
+ 	render() {
+		let archivedMessages = this.props.messages.filter( message => message.isArchived );
+
+		let allMessages = this.props.messages
+			.filter( message =>  !message.isArchived )
+			.concat(archivedMessages);
+
+		let messages = allMessages.map( (message) => {
+			return <Message 
+				key={message.messageID}
+				listID={message.listID} 
+				messageID={message.messageID} 
+				text={message.text} 
+				isArchived={message.isArchived} 
+				listProperties={this.props.listProperties}
+				listName={this.props.listName}
+			/> ;
+		});
+
+		let errorMsg =  (this.state.isError) ? { color: 'red' } : { color: 'black'};
+		let btnState = (this.state.isError) ? true : false;
+		let displayChars = (this.state.displayChars) ? {display: 'inline'} : {display: 'none' };
+
+		return (
+			<div>
+				<h2>{this.props.listName}</h2>
+				<ul>
+					{messages}
+				</ul> 
+				<input onBlur={this.displayCharCount} onKeyDown={this.createMsg} onFocus={this.displayCharCount} onKeyUp={this.charValidation} ref='inputField' type="text" />
+				<button disabled={btnState} onClick={this.createMsg}>Create New Message</button>
+				<br/>
+				<div style={errorMsg}>
+					<span style={displayChars}> {this.state.charCount} </span>
```

Comment:
I think a separate method for blur and another for focus would make it more readable. If you are going to use one action to toggle between 2 states, you have to pay extra effort to find appropriate names to describe what you are doing. e.g. It's easier to understand `hideDisplayChar()` and `showDisplayChar()`. Otherwise you have to use something like `toggleDisplayChar()` which does not directly states what the 2 states are. I got the same feeling with the archiving action on `message-store.js`, I think it's better to at least use the word _toggle_ in this kind of situations.